flow/Trace.cpp

```diff
@@ -220,6 +221,37 @@ struct TraceLog {
 		}
 	};
 
+	struct IssuesList : ITraceLogIssuesReporter, ThreadSafeReferenceCounted<IssuesList> {
+		IssuesList(){};
+		void addIssue(std::string issue) override {
+			MutexHolder h(mutex);
+			issues.insert(issue);
+		}
+
+		void retrieveIssues(std::set<std::string>& out) override {
+			MutexHolder h(mutex);
+			for (auto const& i : issues) {
+				out.insert(i);
+			}
+		}
+
+		void resolveIssue(std::string issue) override {
+			MutexHolder h(mutex);
+			if (issues.find(issue) != issues.end()) {
+				issues.erase(issue);
```

Comment:
I think you can call erase even if the item isn't present, and doing so will reduce the number of lookups by 1. Another option is to instead pass the result of `issues.find` to `erase`, which also eliminates the extra lookup.

Reply:
Done